code/modules/projectiles/gun.dm

```diff
@@ -65,6 +65,8 @@
 	var/list/dispersion = list(0)
 	var/reliability = 100
 
+	var/obj/item/device/firing_pin/pin = /obj/item/device/firing_pin//standard firing pin for most guns.

```

Comment:
This is an interesting way to save a variable. Hrm.

Reply:
I can't think of another way to do it, myself. a firing pin shouldn't be anything but a firing pin.

Reply:
The "proper" way to do this would be:
```dm
var/obj/item/device/firing_pin/pin = null
var/pin_type = obj/item/device/firing_pin
```